### PR TITLE
Hotfix 0.4.0b

### PR DESCRIPTION
**Pull request by awjuliani**
* Corrects observation space description for `PushBlock` environment.
* Fixes bug preventing using environments with python multi-processing.
* Fixes bug preventing agents to be initialized without a brain.

### CODE REVIEW

File: unity-environment/Assets/ML-Agents/Scripts/Agent.cs [hunk 1]
```diff
@@ -257,10 +257,9 @@ public abstract class Agent : MonoBehaviour
         void OnEnable()
         {
             textureArray = new Texture2D[agentParameters.agentCameras.Count];
-            for (int i = 0; i < brain.brainParameters.cameraResolutions.Length; i++)
+            for (int i = 0; i < agentParameters.agentCameras.Count; i++)
             {
-                textureArray[i] = new Texture2D(brain.brainParameters.cameraResolutions[i].width, 
-                    brain.brainParameters.cameraResolutions[i].height, TextureFormat.RGB24, false);
+                textureArray[i] = new Texture2D(1, 1, TextureFormat.RGB24, false);
```
Comment (xiaomaogy) [inline unity-environment/Assets/ML-Agents/Scripts/Agent.cs]: What is the reason behind this change?

File: python/unityagents/rpc_communicator.py [hunk 2]
```diff
@@ -15,7 +15,8 @@
 
 
 class UnityToExternalServicerImplementation(UnityToExternalServicer):
-    parent_conn, child_conn = Pipe()
+    def __init__(self):
+        self.parent_conn, self.child_conn = Pipe()
```
Comment (xiaomaogy) [inline python/unityagents/rpc_communicator.py]: Is this the fix for the bug preventing using environments with python multi-processing.

File: unity-environment/Assets/ML-Agents/Scripts/Agent.cs (hunk 1 above)
Comment (awjuliani) [reply unity-environment/Assets/ML-Agents/Scripts/Agent.cs]: The agent can be initialized without a brain attached, in which case the old code would throw an error. In the new code we use the agent's camera list to initialize the texture2d array. But, since we don't know how big the textures should be, we initialize them to 1x1 to start with.

File: python/unityagents/rpc_communicator.py (hunk 2 above)
Comment (awjuliani) [reply python/unityagents/rpc_communicator.py]: Yep. In the old code they were global variables, which prevented multiple instances of them from being created correctly.

File: unity-environment/Assets/ML-Agents/Scripts/Agent.cs (hunk 1 above)
Comment (xiaomaogy) [reply unity-environment/Assets/ML-Agents/Scripts/Agent.cs]: Why do we want to allow the agent be initialized without a brain attached?

Comment (awjuliani) [reply unity-environment/Assets/ML-Agents/Scripts/Agent.cs]: We may want to programmatically attach brains during runtime, as is the case with `WallJump`.

Comment (xiaomaogy) [reply unity-environment/Assets/ML-Agents/Scripts/Agent.cs]: Oh I see.
